config/config.go

```diff
@@ -32,3 +42,39 @@ func AddFlags(cmd *cobra.Command) {
 		panic(err)
 	}
 }
+
+func PersistentStartupProcessFlags() error {
+
+	if err := convertResolveHostsToMap(); err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func convertResolveHostsToMap() error {
+	for _, resolveHost := range viper.GetStringSlice(optname.Resolve) {
+		split := strings.SplitN(resolveHost, ":", 3)
+		if len(split) != 3 {
+			return fmt.Errorf("invalid resolve host format, expected <hostname>:port:<ip>, got: %s", resolveHost)
+		}
+		host, port, addr := strings.ToLower(split[0]), split[1], split[2]
+		if _, ok := HostToIPResolutionMap[host]; ok {
+			return fmt.Errorf("duplicate hostname specified: %s", host)
+		}
+		IP := net.ParseIP(addr)
+		if IP == nil {
+			return fmt.Errorf("invalid IP address: %s", addr)
+		}
+		if IP.To16() != nil && IP.To4() == nil {
+			return fmt.Errorf("invalid IP address not IPv4 or IPv6: %s", addr)
+		}
```

Comment:
```suggestion
		if IP.To16() == nil && IP.To4() == nil {
			return fmt.Errorf("invalid IP address not IPv4 or IPv6: %s", addr)
		}
```
I think?

Although I don't know why we need this check - if net.ParseIP was successful, we don't need to check whether it's v4 or v6?